modoboa/admin/tests/test_need_dovecot_update.py

```diff
@@ -0,0 +1,93 @@
+
+from django.core.management import call_command
+from django.test import override_settings
+from django.urls import reverse
+from django.db import transaction
+
+from modoboa.core.tests.test_views import SETTINGS_SAMPLE
+
+from modoboa.lib.tests import ModoTestCase
+from .. import factories, models
+
+from modoboa.core import models as core_models
+
+class NeedDovecotUpdateTestCase(ModoTestCase):
+    """Test need dovecot ldap update command."""
+
+    @classmethod
+    def setUpTestData(cls):
+        """Create test data."""
+        super(NeedDovecotUpdateTestCase, cls).setUpTestData()
+        factories.populate_database()
+
+    def setUp(self):
+        """Initiate initial env."""
+        super(NeedDovecotUpdateTestCase, self).setUp()
+        self.workdir = tempfile.mkdtemp()
+        path = "{}/test.com/admin".format(self.workdir)
+        os.makedirs(path)
+
+    def tearDown(self):
+        """Reset test env."""
+        shutil.rmtree(self.workdir)
+
+    def test_need_dovecot_update_init(self):
+        localconfig = core_models.LocalConfig.objects.first()
+        self.assertIsNotNone(localconfig.need_dovecot_update)
+        self.assertFalse(localconfig.need_dovecot_update)
+
+    def test_update_dovecot_update_state_valid_form(self):
+        localconfig = core_models.LocalConfig.objects.first()
+        self.assertFalse(localconfig.need_dovecot_update)
+        url = reverse("core:parameters")
+        settings = SETTINGS_SAMPLE.copy()
+        response = self.client.post(url, settings, format="json")
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), "Parameters saved")
+        localconfig.refresh_from_db()
+        self.assertTrue(localconfig.need_dovecot_update)
+
+    def test_update_dovecot_ldap_conf(self):
+        localconfig = core_models.LocalConfig.objects.first()
+        localconfig.need_dovecot_update = True
+        localconfig.save()
+        self.assertTrue(core_models.LocalConfig.objects.first().need_dovecot_update)
+        
+        tmp_file = os.path.join(self.workdir, "test-dovecot-ldap.conf")
+        self.set_global_parameters({
+            "authentication_type": "ldap",
+            "ldap_dovecot_sync": True,
+            "ldap_dovecot_conf_file": tmp_file,
+            "ldap_server_address": "localhost",
+            "ldap_server_port": "636",
+            "ldap_enable_secondary_server": True,
+            "ldap_secondary_server_address": "localhost2",
+            "ldap_secondary_server_port": "636",
+            "ldap_secured": "ssl",
+            "ldap_bind_dn": "DC=test,DC=lan",
+            "ldap_bind_password": "test",
+            "ldap_search_base": "CN=Users,DC=test,DC=lan",
+            "ldap_search_filter": "(& (objectClass=user) (|(mail=%(user)s)(sAMAccountName=%(user)s)) )"
+        }, app="core")
+
+        # Generated file checks
+        call_command("update_dovecot_conf")
+        self.assertTrue(os.path.exists(tmp_file))
```

Comment:
I think it won't work unless you set `ldap_dovecot_sync` to True...

Reply:
ldap_dovecot_sync is set to True on line 65.

Reply:
`ldap_dovecot_sync` is set to True in previous lines (in `self.set_global_parameters` block)